sql/plan/alter_table.go

```diff
@@ -221,6 +221,9 @@ func (a *AddColumn) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error)
 	if err := a.validateDefaultPosition(tblSch); err != nil {
 		return nil, err
 	}
+	if collatedType, ok := a.column.Type.(sql.TypeWithCollation); ok && collatedType.Collation() == sql.Collation_Invalid {
```

Comment:
Given the use here I think sql.NoCollation is better? Or sql.Collation_None

Reply:
I can understand the preference here, but I think `Collation_Invalid` is the best overall name. If a type uses collations then it requires a valid collation to be used. At this point, the collation has not been _used_, therefore it's still okay. If some code path sets an invalid collation but then doesn't clear it, the program will error due to the invalid collation. Naming it `Collation_None` makes it seem like a valid alternative, as though we're just turning off collations and perhaps using some default behavior. Of course it will still error, but name-wise it seems less clear that it would occur.

Reply:
SGTM